system/lib/webgpu/webgpu.cpp

```diff
@@ -27,3 +38,24 @@ WGPUInstance wgpuCreateInstance(const WGPUInstanceDescriptor* descriptor) {
 
 void wgpuInstanceReference(WGPUInstance) { /* no-op for now */ }
 void wgpuInstanceRelease(WGPUInstance) { /* no-op for now */ }
+
+// WGPUSurface
+
+void wgpuSurfaceGetCapabilities(WGPUSurface surface,
+                                WGPUAdapter adapter,
+                                WGPUSurfaceCapabilities* capabilities) {
+  WGPUTextureFormat preferredFormat =
+      wgpuSurfaceGetPreferredFormat(surface, adapter);
+  assert(preferredFormat == WGPUTextureFormat_BGRA8Unorm ||
+         preferredFormat == WGPUTextureFormat_RGBA8Unorm);
+  capabilities->formatCount = 3;
+  if (preferredFormat == WGPUTextureFormat_RGBA8Unorm) {
+    capabilities->formats = reinterpret_cast<const WGPUTextureFormat*>(
+        kBGRA8UnormPreferredContextFormats.data());
+  } else {
+    capabilities->formats = reinterpret_cast<const WGPUTextureFormat*>(
+        kRGBA8UnormPreferredContextFormats.data());
+  }
+
+  // TODO: What do we return for presentModes and alphaModes?
```

Comment:
presentModes should statically be `[Fifo]`.

alphaModes should statically be one of the following, but I'm not sure which (https://github.com/webgpu-native/webgpu-headers/issues/292):
- `[Auto, Opaque, Premultiplied]`
- `[Opaque, Premultiplied]` I would pick this one for now.